hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -998,7 +1000,16 @@ object HoodieSparkSqlWriter {
       properties.put(HiveSyncConfigHolder.HIVE_SYNC_SCHEMA_STRING_LENGTH_THRESHOLD.key, spark.sessionState.conf.getConf(StaticSQLConf.SCHEMA_STRING_LENGTH_THRESHOLD).toString)
       properties.put(HoodieSyncConfig.META_SYNC_SPARK_VERSION.key, SPARK_VERSION)
       properties.put(HoodieSyncConfig.META_SYNC_USE_FILE_LISTING_FROM_METADATA.key, hoodieConfig.getBoolean(HoodieMetadataConfig.ENABLE))
-
+      try{
+        val passwd = ShimLoader.getHadoopShims.getPassword(spark.sparkContext.hadoopConfiguration, HiveConf.ConfVars.METASTOREPWD.varname)
+        if (passwd != null && !passwd.isEmpty) {
```

Comment:
@ad1happy2go : Can you only use the password from credential store if HiveConf.ConfVars.METASTOREPWD.varname is not set in hadoop configuration ? The current change will have behavior change to existing hive-hudi installations.

Reply:
Thanks @bvaradar for the feedback. I will make that change.

Reply:
Made the changes. Please review.